WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -36,7 +36,7 @@ class PostCoordinator: NSObject {
 
         change(post: post, status: .pushing)
 
-        if mediaCoordinator.isUploadingMedia(for: post) {
+        if mediaCoordinator.isUploadingMedia(for: post) || post.hasFailedMedia {
```

Comment:
So I think this is the cause of the post getting published. Originally, if the post upload fails, we were resetting the `status` to `.draft` so the post will **not be published**. That happens here:

https://github.com/wordpress-mobile/WordPress-iOS/blob/72a217d5b2fec206747e93976eeb323faeb1006f/WordPress/Classes/Services/PostService.m#L259-L268

But since the `upload(post: post)` in line 84 is not getting called anymore, there is no chance for the failure block (above) to ever get called and reset the `status` to `.draft`. 

Also, the condition of `post.hasFailedMedia` may be correct but it does not account for the fact that after this line:

1. We observe for any pending media uploads
2. After all pending media uploads are finished, upload the post again

The problem is, we are only assuming that the media are "pending" and not "failed". In the event that all media are failed, there are no media to **observe and wait for** and so the `self.upload(post: post)` in line 57 is never called again.

Reply:
An idea from @diegoreymendez: We can probably merge these:

https://github.com/wordpress-mobile/WordPress-iOS/blob/72a217d5b2fec206747e93976eeb323faeb1006f/WordPress/Classes/Services/PostCoordinator.swift#L29

https://github.com/wordpress-mobile/WordPress-iOS/blob/72a217d5b2fec206747e93976eeb323faeb1006f/WordPress/Classes/Services/PostCoordinator.swift#L149

...since `retrySave()` does the restarting of failed media before uploading the post.